ledger/eval.go

```diff
@@ -859,17 +859,22 @@ func (eval *BlockEvaluator) transaction(txn transactions.SignedTxn, evalParams *
 			continue
 		}
 
-		dataNew := data.WithUpdatedRewards(eval.proto, rewardlvl)
-		effectiveMinBalance := dataNew.MinBalance(&eval.proto)
-		if dataNew.MicroAlgos.Raw < effectiveMinBalance.Raw {
-			return fmt.Errorf("transaction %v: account %v balance %d below min %d (%d assets)",
-				txid, addr, dataNew.MicroAlgos.Raw, effectiveMinBalance.Raw, len(dataNew.Assets))
-		}
+		// Only do those checks if needed. It is useful to skip them if the user cannot provide
+		// account data that contains enough information to compute the correct minimum balance
+		// (the case with indexer).
+		if eval.validate || eval.generate {
```

Comment:
Please avoid the indentation to match the rest of the code by changing this to

```
if !eval.validate && !eval.generate {
    continue
}
```

Reply:
I think that we should take one step back here. 

This entire for loop could be skipped in case `!eval.validate && !eval.generate` holds true. 
Please take a close look, but if this true, I think that we could extract this for-loop into a separate method. That would address the indentation issue, provide faster code and work for the indexer - all at the same time.

Reply:
You are right, thanks!